cranelift/codegen/src/isa/x64/inst/mod.rs

```diff
@@ -57,17 +66,24 @@ pub enum Inst {
         loc: SourceLoc,
     },
 
+    /// The high result of a (un)signed multiply: imul/mul using RAX:RDX.
+    MulHi { size: u8, signed: bool, rhs: RegMem },
+
     /// A synthetic sequence to implement the right inline checks for remainder and division,
-    /// assuming the dividend is in $rax.
-    /// Puts the result back into $rax if is_div, $rdx if !is_div, to mimic what the div
+    /// assuming the dividend is in %rax.
+    /// Puts the result back into %rax if is_div, %rdx if !is_div, to mimic what the div
     /// instruction does.
     /// The generated code sequence is described in the emit's function match arm for this
     /// instruction.
+    ///
+    /// Note: %rdx is marked as modified by this instruction, to avoid an early clobber problem
+    /// with the temporary and divisor. Make sure to zero %rdx right before this instruction!
     CheckedDivOrRemSeq {
         is_div: bool,
         is_signed: bool,
```

Comment:
Drive-by / pre-existing, but perhaps we could turn `is_div` and `is_signed` into their own (combined) enum -- SDiv, UDiv, SRem, URem?

Reply:
Also .. for "Make sure to zero %rdx right before this instruction!" please add a line explaining why this is necessary and what will (or might) happen if you don't.